### PR TITLE
Task2.2-2.3

### PR DESCRIPTION
**Pull request by Nikavar**
finished task2.2. _ login to the game store with token

### CODE REVIEW

File: GameStore.Service/AccountService.cs [hunk 1]
```diff
@@ -25,6 +30,17 @@ public AccountService(IAccountRepository accountRepository, IUnitOfWork unitOfWo
             this.mapper = mapper;
         }
 
+        public async Task<Account> LoginAccountAsync(string username, string password)
+        {
+            if(string.IsNullOrEmpty(username) && string.IsNullOrEmpty(password))
+            {
+                throw new NullReferenceException();
```
Comment (ggabunia) [inline GameStore.Service/AccountService.cs]: NullReferenceException is not really correct. Throw someting else, e.g. NotFoundException  or create a custom one.

File: GameStore.Service/AccountService.cs [hunk 2]
```diff
@@ -25,6 +30,17 @@ public AccountService(IAccountRepository accountRepository, IUnitOfWork unitOfWo
...
+            }  
+                
+            return await accountRepository.LoginAccountAsync(username, password);
```
Comment (ggabunia) [inline GameStore.Service/AccountService.cs]: login logic should not be in repository. Retrieve user by email, than checks password's hash against saved password for the user. If they re the same, authorize the user.

File: GameStore/Controllers/AccountController.cs [hunk 3]
```diff
@@ -28,5 +32,19 @@ public async Task<ActionResult> RegisterAccount([FromBody] AccountModel model)
 
             return BadRequest(Warnings.AccountAlreadyExists<Account>());
         }
+
+        [HttpPost]
+        public async Task<ActionResult> LoginAccount(string username, string password)
```
Comment (ggabunia) [inline GameStore/Controllers/AccountController.cs]: You are potentially throwing an exception in the service. Add try/catch to handle that case.

Comment (ggabunia) [inline GameStore/Controllers/AccountController.cs]: Add required attributes for the parameters

Comment (Nikavar) [reply GameStore/Controllers/AccountController.cs]: [HttpPost]
        public async Task<ActionResult> LoginAccount(string username, string password)
        {
            try
            {
                var result = await accountService.LoginAccountAsync(username, password);

                if (result != null)
                {
                    var token = Helper.TokenGeneration(result, configuration);
                    HttpContext.Response.Cookies.Append("Token", token);
                    return Ok(result);
                }
            }

            catch (Exception ex)
            {
                return BadRequest(ex.Message);
            }

            return BadRequest("your username or/and password is wrong!");
        }

File: GameStore.Service/AccountService.cs (hunk 2 above)
Comment (Nikavar) [reply GameStore.Service/AccountService.cs]: public async Task<Account?> LoginAccountAsync(string username, string password)
        {
            if(string.IsNullOrEmpty(username) && string.IsNullOrEmpty(password))
            {
                throw new Exception("username or/and password is null or empty!");
            }

            password = ComputeSha256Hash(password);
            var entity = await accountRepository.GetManyAsync(x => x.UserName == username && x.Password == password);

            return entity.FirstOrDefault();                 
        }

File: GameStore.Service/AccountService.cs (hunk 1 above)
Comment (Nikavar) [reply GameStore.Service/AccountService.cs]: if(string.IsNullOrEmpty(username) && string.IsNullOrEmpty(password))
            {
                throw new Exception("username or/and password is null or empty!");
            }

File: GameStore/Controllers/AccountController.cs (hunk 3 above)
Comment (Nikavar) [reply GameStore/Controllers/AccountController.cs]: [HttpPost]
        public async Task<ActionResult> LoginAccount([FromQuery] string username, string password)

**Pull request merged**